### PR TITLE
LPS-198500 Adapting openSimpleInputModal to render textarea inputs

### PR DESCRIPTION
**Pull request by Stephy6**
Motivation
=======
The motivation behind this PR is the modification of the openInputSimpleModal component (1st commit) so it is able to render textarea inputs. The component receives a boolean parameter to know if the desired input is textare or standard input.  What do you think?
https://github.com/liferay-echo/liferay-portal/pull/14043

Best regards.

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/components/SimpleInputModal.es.js
Comment (markocikos) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/components/SimpleInputModal.es.js:154]: Nice choice, changing implementation to Clay! :+1: 

We can future-proof new `SimpleInputModal` API by just forwarding new prop to the [Clay component](https://github.com/liferay/clay/blob/master/packages/clay-form/src/Input.tsx#L162). Something like:
```js
<ClayInput
    component={mainFieldComponent}
>
```
New prop would be `mainFieldComponent`, to be consistent with existing naming pattern. In usages, you can set `mainFieldComponent` to `textarea`, if needed. I think you don't even need to set default here, because Clay [sets default to input](https://github.com/liferay/clay/blob/master/packages/clay-form/src/Input.tsx#L184), if `component` is unset.

**Review by markocikos — CHANGES_REQUESTED** [review]
@Stephy6 Sorry for a bit late review! Please see comment inline. I only looked at `frontend-js-web` part of the PR.

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/components/SimpleInputModal.es.js
Comment (Stephy6) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/components/SimpleInputModal.es.js:154]: Oh I see, thank you

**Review by markocikos — APPROVED** [review]
@Stephy6 `frontend-js-web` part LGTM! :sparkles:

**Comment by markocikos** [comment]
@Stephy6 if you're fine with the solution, feel free to `ci : forward` this, or reopen on echo's fork and forward / QA there.

**Comment by Stephy6** [comment]
> @Stephy6 if you're fine with the solution, feel free to `ci : forward` this, or reopen on echo's fork and forward / QA there.

I'm waiting for an additional review by the workflow team to review, I already let them know, thank you!

**Comment by ealonso** [comment]
ci:forward

**Comment by cgmonte** [comment]
Hey @Stephy6, thanks for removing another bit of AUI from Workflow. Because you modified Java files on the Workflow side, I prefer to also have a back-end developer taking a look. I will request that on our thread in Slack.

About the purely front-end side of things, I have no observations regarding the implementation, it looks good to me, but I think it would be better to align modal to the center of the screen, for the sake of consistency, as we just came from a centered modal dialog when we assigned the task, see:

Nov06_120739

and then:

Nov06_120821

That would take another change to [OpenSimpleInputModal.es.js](https://github.com/liferay-frontend/liferay-portal/pull/3764/files#diff-5034d13d7e6560229e04256e73e0f4b86d28f84b08f0a16f0b6f7bdaaa3f8da9), as it does not exposes a `center` property like (ClayModal [does](https://clayui.com/docs/components/modal/api.html)).

For the same reason, we should consider making both modals the same dimensions.

What do you think, @markocikos? Beyond consistency, I am referencing to this [post of yours](https://liferay.slack.com/archives/CNBG06JS3/p1621431675004400?thread_ts=1621427237.003100&cid=CNBG06JS3) in Slack, where your brought up Lexicon:

> If the modal is fixed and doesn’t grow while it is being used, we recommend placing the modal vertically and horizontally centered to call users attention. This placement of the modal feels more intuitive and lets users know that they must complete an action before continuing with the task.

**Comment by Stephy6** [comment]
> Collaborator

Understood about the alignment part  @markocikos , what about the dimensions that @cgmonte also mentioned above ? Should we add another prop to indicate the size of the modal ?

**Comment by markocikos** [comment]
Yeah, makes sense to expose `size` as well :+1:

**Comment by Stephy6** [comment]
> Yeah, makes sense to expose `size` as well 👍

Great, just updated, thanks!

**Comment by cgmonte** [comment]
Just tested it and it is perfect, @Stephy6! LGTM too!

**Comment by cgmonte** [comment]
Unrelated question: clean deploying `frontend-js-web` and `portal-workflow-task-web` wasn't enough to see the results and I had to antall it. What module did I miss to deploy? I could never see `modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/components/SimpleInputModal.es.js` updated in devtools source tab, even after cleaning cache etc.

**Comment by Stephy6** [comment]
ci:forward

**Comment by guilhermedcamacho** [comment]
Hey @Stephy6, the back-end changes looks good to me :+1:  Thanks!

**Comment by Stephy6** [comment]
> Hey @Stephy6, the back-end changes looks good to me 👍 Thanks!

Hey @guilhermedcamacho ! since the relevant tests are failing could you assign a QA member to test this before force-forwarding it? Thanks in advance

**Comment by Luizcarlosqueiroz** [comment]
ci:test:relevant

**Comment by patriciajeanperez** [comment]
Hey @Stephy6, can you please take a look at these issues before we forward? Thank you!

1. The modals in `My Workflow Tasks` don't seem to be consistent. You can use the screenshots @cgmonte attached above. The `Approve` & `Reject` modals now have the Save button, while the `Assign to Me` modal still has the Done button. Is this an oversight or something that we will just address in the future?

2. The Comment text area for approving and rejecting a workflow is now a required field. Is this intended? If it is, we will need to update our documentations to reflect that.

3. There is a minor bug when adding a comment to a pending asset.

- Assign Single Approver to Blogs
- Add a Blog Entry
- Go to My Workflow Tasks - Assigned to My Roles
- Click on ellipsis next to Blog Entry - Assign to Me
- Go to Assigned to Me
- Click on Blog Entry
- Add comment

Error message displays. Navigating back to entry will still display the comment. 

Let me know if you have any questions. Thanks again!

**Comment by cgmonte** [comment]
Good catches, @patriciajeanperez

**Comment by markocikos** [comment]
@Stephy6 It's ok to extend API of `openSimpleInputModal` with any props that mirror `ClayModal` or `ClayInput` props.

This would likely be too much for the scope of this task, but you may also consider refactoring `openSimpleInputModal` to use  `openModal` [utility](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js) instead of `ClayModal`. Some issues were fixed in it, and some DXP-specifics added.

**Comment by Stephy6** [comment]
> @Stephy6 It's ok to extend API of `openSimpleInputModal` with any props that mirror `ClayModal` or `ClayInput` props.
> 
> This would likely be too much for the scope of this task, but you may also consider refactoring `openSimpleInputModal` to use `openModal` [utility](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js) instead of `ClayModal`. Some issues were fixed in it, and some DXP-specifics added.

I'll be glad to do the refactor but yes, outside of this task. If you want to create a ticket and assign it to me.

**Comment by Stephy6** [comment]
Hi @patriciajeanperez , changes applied, if you want to test it again it would be awesome. Thanks in advance!

**Comment by patriciajeanperez** [comment]
Thank you, @Stephy6! I'll do that

**Comment by patriciajeanperez** [comment]
ci:test:relevant

**Comment by patriciajeanperez** [comment]
Resent here with test fixes: https://github.com/liferay-frontend/liferay-portal/pull/3831 @Stephy6

**Comment by patriciajeanperez** [comment]
ci:close

**Pull request closed without merging**
